### PR TITLE
DUOS-762[risk=no] Added Cancel Button to New Chair Console

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-762](https://broadworkbench.atlassian.net/browse/DUOS-762)

PR aims to add Cancel button to the New Chair Console. Code also splits up some helper methods in NewChairConsole so that it can be used in both create and cancel workflows. Code also refactors legacy code as a result of returning res.json on the createDARElection ajax method, which is used in NewChairConsole. Finally, code adds minor templating for Vote/Final conditional rendering.

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] PR is labeled with a Jira ticket number and includes a link to the ticket
- [x] PR is labeled with a security risk modifier [no, low, medium, high] 
- [x] PR describes scope of changes

In all cases:

- [x] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [x] Get PO sign-off for all non-trivial UI or workflow changes
- [x] Verify all tests go green
- [x] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/NewChairConsole.js [hunk 1]
```diff
@@ -39,26 +39,55 @@ const getElectionDate = (election) => {
...
+
     if (!isNil(e)) {
       switch (e.status) {
-        case "Open" :
-          const votes = filter({type: "DAC", dacUserId: Storage.getCurrentUser().dacUserId})(electionInfo.votes);
+        case 'Open' :
+          const votes = filter({type: 'DAC', dacUserId: currentUserId})(electionInfo.votes);
+          //Votes can have timestamps stored anywhere between these four key atttributes (result of older data)
+          //Current votes (02/18/2021) store timestamps on updateDate
+          const isFinal = !isEmpty(votes.find(vote => !isNil(vote.createDate) || !isNil(vote.updateDate) || !isNil(vote.lastUpdate) || !isNil(vote.lastUpdateDate)));
```
Comment (rushtong) [inline src/pages/NewChairConsole.js:75]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/DataBiosphere/duos-ui/pullRequest?prid=6975233)
```suggestion
          const isFinal = !isEmpty(votes.find((vote) => !isNil(vote.createDate) || !isNil(vote.updateDate) || !isNil(vote.lastUpdate) || !isNil(vote.lastUpdateDate)));
```

**Review by rushtong — COMMENTED** [review]
This functions great! I am confused about the implementation of side-by-side buttons:

Screen Shot 2021-02-19 at 12 17 57 PM

The invision spec shows a cancel icon, not a button:

Screen Shot 2021-02-19 at 12 24 49 PM

I don't see any discussion of this in the associated story. Can we touch base on this and maybe get feedback from @solideoglori on how we chose to implement the differences?

**Comment by rushtong** [comment]
The codacy checks here are invalid and should be ignored.

**Comment by JVThomas** [comment]
> This functions great! I am confused about the implementation of side-by-side buttons:
> 
> Screen Shot 2021-02-19 at 12 17 57 PM
> 
> The invision spec shows a cancel icon, not a button:
> 
> Screen Shot 2021-02-19 at 12 24 49 PM
> 
> I don't see any discussion of this in the associated story. Can we touch base on this and maybe get feedback from @solideoglori on how we chose to implement the differences?

They're all supposed to be icon/buttons with custom styles. I'd figured it'll be best to get the functionality done now and implement the styling in another PR once it's all set up.

**Comment by rushtong** [comment]
> They're all supposed to be icon/buttons with custom styles. I'd figured it'll be best to get the functionality done now and implement the styling in another PR once it's all set up.

Ah, gotcha. In that case, 👍🏽

**Review by rushtong — APPROVED** [review]

**Comment by JVThomas** [comment]
Leaving a note that the current npm audit flag (regarding immer dependency in react-scripts) is a false positive
https://github.com/facebook/create-react-app/pull/10412#issuecomment-781391257

Vulnerability regarding react-responsive-navbar has been handled by DUOS-810 https://github.com/DataBiosphere/duos-ui/pull/885

**Pull request merged**